src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java

```diff
@@ -505,4 +510,91 @@ void shouldReturnTransactionsCorrectlyForDateRangesAndNumberOfTransactions() {
+                    .withReference("ref-2")
+                    .withEmail("test2@email.com")
+                    .withCardholderName("Jane D")
+                    .withTransactionDetails(transactionDetails2.toString())
+                    .insert(rule.getJdbi())
+                    .toEntity();
+
+            transactionDao.redactPIIFromTransaction(transactionToRedact.getExternalId());
+
+            TransactionEntity transactionEntityRedacted = transactionDao.findTransactionByExternalId(transactionToRedact.getExternalId()).get();
+
+            assertThat(transactionEntityRedacted.getReference(), is("<REDACTED>"));
+            assertThat(transactionEntityRedacted.getCardholderName(), is("<REDACTED>"));
+            assertThat(transactionEntityRedacted.getEmail(), is("<REDACTED>"));
+
+            JsonNode jsonNode = objectMapper.readTree(transactionEntityRedacted.getTransactionDetails());
+
+            assertThat(jsonNode.get("reference"), is(Matchers.nullValue()));
+            assertThat(jsonNode.get("cardholder_name"), is(nullValue()));
+            assertThat(jsonNode.get("email"), is(nullValue()));
+            assertThat(jsonNode.get("address_line1").asText(), is("<REDACTED>"));
+            assertThat(jsonNode.get("address_line2").asText(), is("<REDACTED>"));
+
+            TransactionEntity transactionEntityNotRedacted = transactionDao.findTransactionByExternalId(transactionThatShouldNotBeRedacted.getExternalId()).get();
+            assertThat(transactionEntityNotRedacted.getReference(), is("ref-2"));
+            assertThat(transactionEntityNotRedacted.getCardholderName(), is("Jane D"));
+            assertThat(transactionEntityNotRedacted.getEmail(), is("test2@email.com"));
+        }
+
+        @Test
+        void shouldNotReplaceNonMandatoryFieldsWithARedactedStringWhenBlank() throws JsonProcessingException {
+            JsonObject transactionDetails1 = new JsonObject();
+            transactionDetails1.addProperty("reference", "ref-1");
+
+            TransactionEntity transactionToRedact = aTransactionFixture()
+                    .withReference("ref-1")
+                    .withEmail(null)
+                    .withCardholderName(null)
+                    .withTransactionDetails(transactionDetails1.toString())
+                    .insert(rule.getJdbi())
+                    .toEntity();
+
+            transactionDao.redactPIIFromTransaction(transactionToRedact.getExternalId());
+
+            TransactionEntity transactionEntity = transactionDao.findTransactionByExternalId(transactionToRedact.getExternalId()).get();
+
+            assertThat(transactionEntity.getReference(), is("<REDACTED>"));
+            assertThat(transactionEntity.getCardholderName(), is(nullValue()));
+            assertThat(transactionEntity.getEmail(), is(nullValue()));
+
+            JsonNode jsonNode = objectMapper.readTree(transactionEntity.getTransactionDetails());
+
+            assertThat(jsonNode.get("reference"), is(nullValue()));
```

Comment:
Why is this reference null when it was added above?
```
            JsonObject transactionDetails1 = new JsonObject();
            transactionDetails1.addProperty("reference", "ref-1");
```

Reply:
This is because the field is being removed from transaction_details JSON 

as mentioned in PR desc

> Removed fields reference, cardholder_name, email from transaction_details JSON as these are not used anywhere and are also top-level DB columns. So having them in transaction_details as <REDACTED> doesn't add any value